### ISSUE TITLE
training label issue

### ISSUE THREAD

**Issue by zhgjia** [TianLin0509/DNN_detection_via_keras#3]
why  is the size of label 16 instead of payloadBits_per_OFDM（128）？Is there a difference between the two?

**Comment by TianLin0509** [comment]
sorry, I'm not understanding your problem

**Comment by zhgjia** [comment]
What I mean is that there are 128 Numbers of a OFDM symbol (64-IFFT) with real and image parts. Why is the predicted label only 16 of them? Will this affect the simulation results of bit error rate?




------------------ 原始邮件 ------------------
发件人: "Lin Tian"<notifications@github.com>;
发送时间: 2019年7月6日(星期六) 中午12:51
收件人: "TianLin0509/DNN_detection_via_keras"<DNN_detection_via_keras@noreply.github.com>;
抄送: "张佳"<2673224202@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [TianLin0509/DNN_detection_via_keras] training label issue (#3)




sorry, I'm not understanding your problem

**Comment by TianLin0509** [comment]
This is because that the author said: he used totally 8 networks to predict the 128bits. And therefore, every network only need to predict 16bits. Please kindly refer to the paper. Since the 8 network intutively have similar performance, in the program, we can only consider one network and predict 16bits. You can choose to predict 128bits, it will work, but results in some performance loss, as expected.

**Comment by zhgjia** [comment]
Ok, I get it. Thank you very much!


------------------ 原始邮件 ------------------
发件人: "Lin Tian"<notifications@github.com>;
发送时间: 2019年7月6日(星期六) 下午2:22
收件人: "TianLin0509/DNN_detection_via_keras"<DNN_detection_via_keras@noreply.github.com>;
抄送: "张佳"<2673224202@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [TianLin0509/DNN_detection_via_keras] training label issue (#3)




This is because that the author said: he used totally 8 networks to predict the 128bits. And therefore, every network only need to predict 16bits. Please kindly refer to the paper. Since the 8 network intutively have similar performance, in the program, we can only consider one network and predict 16bits. You can choose to predict 128bits, it will work, but results in some performance loss, as expected.

**Comment by BassantTolba1234** [comment]
Dear Sir,
please I need to know which type of modulation scheme is used here QPSK as mentioned in the paper or QAM as written in the code ? what should I do to make it 16 QAM ?
the second question , can you kindly share the code required to get the same results figures ?
thanks in advance

**Comment by haide666** [comment]
Since some subcarriers locate in the channel with low SNR, I might not agree with the point that every network predicting 16bits had the similar performance. For example, in the low pass channel with AWGN, last subcarriers will have lower SNRs than those of the first subcarriers. So the bits modulated in the last subcarriers  will be more difficult to be predicted.

**Comment by BassantTolba1234** [comment]
Dear Sir,
I really appreciate your hard work..and please I have a question, in main file line 48, why do you make (model.evaluate) on function called (validation_gen) which exactly contains the generation of training datasets not the test datasets??..

why did not you apply evaluation the model on testing datasets you provided , instead of training datasets ?
I'm waiting for your reply..
thanks in advance.

**Comment by BassantTolba1234** [comment]
Dear sir, please what is the value of SNR (Signal Noise to ratio used in training the model ) ??
